include/domain_bridge/parse_domain_bridge_yaml_config.hpp

```diff
@@ -33,42 +33,46 @@ class YamlParsingError : public std::runtime_error
   {}
 };
 
-/// Create a DomainBridge from a YAML file.
+/// Create a DomainBridgeConfig object from a YAML file.
 /**
  * The YAML file may contain the following optional keys:
  *
  * - name: Name of the domain bridge
+ * - from_domain: The default 'from_domain' used for bridged topics.
+ *     If omitted, then each topic bridge must specify it's own value.
+ * - to_domain: The default 'to_domain' used for bridged topics
+ *     If omitted, then each topic bridge must specify it's own value.
  * - topics: A map of topic names to a map of topic bridge information
  *
- * For each topic bridge, the follow information is required, in the form of YAML keys:
+ * For each topic bridge, the follow keys are valid:
  *
- * - type: Message type to bridge
- * - from_domain: Subscribe to the topic on this domain ID
- * - to_domain: Publish to the topic on this domain ID
+ * - type: Message type to bridge (required)
+ * - from_domain: Subscribe to the topic on this domain ID (optional)
+ * - to_domain: Publish to the topic on this domain ID (optional)
  *
  * For example,
  *
  * ```yaml
  * name: my_bridge
+ * from_domain: 2
+ * to_domain: 3
  * topics:
  *   chatter:
  *     type: example_interfaces/msg/String
- *     from_domain: 2
- *     to_domain: 3
  *   clock:
  *     type: rosgraph_msgs/msg/Clock
- *     from_domain: 2
- *     to_domain: 3
+ *     # Override the default 'to_domain' above
+ *     to_domain: 4
  * ```
  *
  * \param file_path: Path to the YAML file.
- * \return A pointer to an initialized DomainBridge that was configured based on input YAML file.
+ * \return A DomainBridgeConfig object populated according to the input YAML file.
  * \throws domain_bridge::YamlParsingError if there was an error parsing the YAML file,
  *   for example, if a required key is missing.
  */
 DOMAIN_BRIDGE_PUBLIC
-DomainBridge domain_bridge_from_yaml(std::filesystem::path file_path);
+DomainBridgeConfig parse_domain_bridge_yaml_config(std::filesystem::path file_path);
```

Comment:
Could also be a static method in a DomainBridgeYamlParser, like DomainBridgeYamlParser::parse().

Reply:
I went with a free function since it isn't really stateful; what advantage does a `DomainBridgeYamlParser` object (or static method) have over this free function?

Reply:
No advantage in terms of functionality. More along the lines of ensuring that we're all using a consistent approach. I have no issue with the particular name, per se, as long as similar kinds of functions are named in a similar way and don't use a different approach, like an object/static method. Personally, I would've thought of a parser object, as I mentioned, but there is no intrinsic benefit to that. Just making sure about the convention. 

Also, the consistency is for the reader of the code. It's nice, IMO, if there are predictable naming conventions.

Reply:
I agree, we should be consistent. I'm still a little confused as to what we could change here to be consistent with other ROS 2 packages, though. Most free function names I see are actions, [for example](https://github.com/ros2/rclcpp/blob/35c89c8afcac89a43872f0bf5b763ff324c82b05/rclcpp/include/rclcpp/utilities.hpp). Perhaps renaming the header/cpp files as you suggest would help?

Reply:
Sorry, my fault for not being clear. I didn't mean that the code necessarily had to change, I just wanted to ensure that we were indeed using a consistent convention and that this was it. I raised the point because it's not what I would have done and wanted to check on this.

Reply:
Gotcha. TBH, I picked the name sort-of arbitrarily. I'm not sure that we have an explicit convention written down anywhere. For style, we refer to the [Google style guide](https://google.github.io/styleguide/cppguide.html) (with some exceptions), and I didn't notice anything in there about this topic.